packages/@aws-cdk/aws-glue/lib/connection.ts

```diff
@@ -0,0 +1,130 @@
+import { Construct, IResource, Resource, Aws } from '@aws-cdk/core';
+import ec2 = require('@aws-cdk/aws-ec2');
+import { CfnConnection } from './glue.generated';
+
+export enum ConnectionInputTypes {
+  /**
+   * The type of the connection. Currently, only JDBC is supported; SFTP is not supported.
+   */
+  JDBC = 'JDBC',
+}
+
+
+export interface IPhysicalRequirements {
+  /**
+   * @attribute
+   */
+  readonly availabilityZone?: string;
+
+  /**
+   * @attribute
+   */
+  readonly securityGroupIds?: ec2.ISecurityGroup[];
+
+  /**
+   * @attribute
+   */
+  readonly subnet?: ec2.ISubnet;
+}
+
+export interface IConnectionInput {
```

Comment:
The ergonomics of this interface leave a few things to be desired (for example, `type` requires you to specify an enum that has a single value).